### PR TITLE
Improve diagnostics serialization behaviour

### PR DESCRIPTION
**Pull request by X9VoiD**
Make a "best-effort" diagnostics output in case of serialization errors. The title is also changed to "prompt" the user better that they are supposed to choose a location for saving the diagnostics.

## Changes

- "Exporting diagnostic information" -> "Save diagnostic information to..."
- Handle serialization exceptions

### CODE REVIEW

File: OpenTabletDriver.Desktop/Diagnostics/DiagnosticInfo.cs [hunk 1]
```diff
@@ -35,6 +37,12 @@ private static string GetAppVersion()
             return "OpenTabletDriver v" + Assembly.GetEntryAssembly().GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
         }
 
+        [OnError]
+        internal void OnError(StreamingContext _, ErrorContext errorContext)
+        {
+            errorContext.Handled = true;
```
Comment (InfinityGhost) [inline OpenTabletDriver.Desktop/Diagnostics/DiagnosticInfo.cs:43]: Maybe worth logging the error?

Comment (X9VoiD) [reply OpenTabletDriver.Desktop/Diagnostics/DiagnosticInfo.cs:44]: Added.

**Review by gonX — APPROVED** [review]
LGTM, should probably add this to the 0.6.x milestone

**Comment by X9VoiD** [comment]
Rebased.

**Review by gonX — APPROVED** [review]

**Review by InfinityGhost — APPROVED** [review]

**Pull request merged**